### PR TITLE
fix(qwik): Fix Symbol bindings

### PR DESCRIPTION
**Pull request by mhevery**
- Symbols did not support bindings from parent
- It was not possible to have client based Components
- Created `Image` as a client side component so that it can render
  correctly under different bindings.

### CODE REVIEW

File: packages/core/src/generators/qwik/component.ts [hunk 1]
```diff
@@ -82,7 +84,10 @@ export function addComponent(
   opts: { isRoot?: boolean; shareStyles?: boolean } = {},
 ) {
   const _opts = { isRoot: false, shareStyles: false, ...opts };
-  compileAwayBuilderComponentsFromTree(component, compileAwayComponents);
+  compileAwayBuilderComponentsFromTree(
+    component,
+    omit(compileAwayComponents, ['Image']),
+  );
```
Comment (samijaber) [inline packages/core/src/generators/qwik/component.ts:90]: I am slightly weary of pulling in `lodash` dependencies given how notorious it is for increasing bundle size.

For a relatively simple use-case like this one, how do you feel about omitting using the spread operator?

```suggestion
  const { Image, ...compileAwayComponentsWithoutImage } = compileAwayComponents
  compileAwayBuilderComponentsFromTree(
    component,
    compileAwayComponentsWithoutImage,
  );
```

File: packages/core/src/plugins/compile-away-builder-components.ts [hunk 2]
```diff
@@ -573,6 +563,29 @@ export const compileAwayBuilderComponents = (
...
 
+function updateQueryParam(uri = '', key: string, value: string) {
+  const re = new RegExp('([?&])' + key + '=.*?(&|$)', 'i');
+  const separator = uri.indexOf('?') !== -1 ? '&' : '?';
+  if (uri.match(re)) {
+    return uri.replace(re, '$1' + key + '=' + encodeURIComponent(value) + '$2');
+  }
+
+  return uri + separator + key + '=' + encodeURIComponent(value);
+}
```
Comment (samijaber) [inline packages/core/src/plugins/compile-away-builder-components.ts:574]: Curious why `updateQueryParam` is duplicated here and in `qwik/directives.ts`. They look identical to me. 

What do you think of sharing the helper between both places to reduce duplication?

**Review by samijaber — APPROVED** [review]

File: packages/core/src/symbols/symbol-processor.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+const createBuilderElement = (
+  options?: Partial<BuilderElement>,
+): BuilderElement => ({
+  '@type': '@builder.io/sdk:Element',
+  id:
+    'builder-' +
+    Math.random()
+      .toString(36)
+      .split('.')[1],
```
[the remark this answers is not in the document]
Comment (mhevery) [reply packages/core/src/symbols/symbol-processor.ts:165]: fixed

File: packages/core/src/plugins/compile-away-builder-components.ts (hunk 2 above)
Comment (samijaber) [reply packages/core/src/plugins/compile-away-builder-components.ts:574]: I see, makes sense. Thanks for explaining 👍🏽

**Pull request merged**